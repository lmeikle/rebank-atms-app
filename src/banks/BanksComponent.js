import React from 'react';
import { Link } from 'react-router-dom';
import { string } from 'prop-types';
import logo from './icon.svg';
import './Banks.css';

/**
 * Stateless component which renders a single Bank component.
 * Clicking on it will show a list of closest ATM's for the selected bank.
 */
const BanksComponent = ({ name, url }) => (
  <Link to={{ pathname: '/atms', state: { name, url } }}>
    <div className="BanksContainer">
      <img src={logo} className="BanksLogo" alt="logo" />
      <div>{name}</div>
    </div>
  </Link>
);

BanksComponent.propTypes = {
  name: string.isRequired,
  url: string.isRequired
};

BanksComponent.defaultProps = {};

export default BanksComponent;
