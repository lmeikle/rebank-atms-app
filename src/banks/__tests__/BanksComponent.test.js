import React from 'react';
import { shallow } from 'enzyme';
import BanksComponent from '../BanksComponent';

let props = {
  name: 'Barclays Bank',
  url: 'https://www.example.com'
};

const banksComponent = shallow(<BanksComponent {...props} />);

test('renders correctly', () => {
  expect(banksComponent).toMatchSnapshot();
});
