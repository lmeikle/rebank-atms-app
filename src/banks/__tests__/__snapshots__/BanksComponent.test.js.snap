// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BanksComponent
    name="Barclays Bank"
    url="https://www.example.com"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="banks"
>
        <img
                alt="logo"
                className="banks-logo"
                src="icon.svg"
        />
        <div>
                Barclays Bank
        </div>
</div>,
      "replace": false,
      "to": Object {
        "pathname": "/atms",
        "state": Object {
          "name": "Barclays Bank",
          "url": "https://www.example.com",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="banks-logo"
            src="icon.svg"
/>,
          <div>
            Barclays Bank
</div>,
        ],
        "className": "banks",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "banks-logo",
            "src": "icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Barclays Bank",
          },
          "ref": null,
          "rendered": "Barclays Bank",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="banks"
>
          <img
                    alt="logo"
                    className="banks-logo"
                    src="icon.svg"
          />
          <div>
                    Barclays Bank
          </div>
</div>,
        "replace": false,
        "to": Object {
          "pathname": "/atms",
          "state": Object {
            "name": "Barclays Bank",
            "url": "https://www.example.com",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="banks-logo"
              src="icon.svg"
/>,
            <div>
              Barclays Bank
</div>,
          ],
          "className": "banks",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "banks-logo",
              "src": "icon.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Barclays Bank",
            },
            "ref": null,
            "rendered": "Barclays Bank",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
